packages/dashboard/src/components/interfaceSkeleton/index.js

```diff
@@ -81,33 +81,24 @@ const InterfaceSkeleton = ({ additionalRoutes }) => {
     });
   }, [addInitialFetchListener, push, currentPath]);
 
-  const fullPath = useMemo(() => {
-    return currentPath.includes(APP_ROUTES.TEMPLATE_DETAIL) &&
-      templateId &&
-      currentTemplate
-      ? `${currentPath}/${templateId}`
-      : currentPath;
-    // Disable reason: avoid sending duplicate tracking events.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [currentPath, currentTemplate]);
-
   useEffect(() => {
     if (!isRedirectComplete) {
       return;
     }
-
-    if (currentPath.includes(APP_ROUTES.TEMPLATE_DETAIL) && !currentTemplate) {
+    // backwards compatibility for template details pages,
+    // if templateId is present modal will open to appropriate template
+    if (currentPath.includes(NESTED_APP_ROUTES.TEMPLATES_GALLERY_DETAIL)) {
+      replace(
+        templateId
+          ? `${APP_ROUTES.TEMPLATES_GALLERY}?id=${templateId}&isLocal=${
+              isLocal || false
+            }`
+          : `${APP_ROUTES.TEMPLATES_GALLERY}`
```

Comment:
💭 A small idea but if the id that's entered doesn't exist, I think we should clear the url? Could be misleading? 
![image](https://user-images.githubusercontent.com/22185279/147679677-995c4e83-b760-4454-bc66-f4428f1f6d48.png)

Reply:
Oh yeah great idea!